presto-main/src/main/java/com/facebook/presto/util/Types.java

```diff
@@ -32,4 +32,15 @@ private Types() {}
                 value.getClass().getName());
         return target.cast(value);
     }
+
+    /**
+     * Uses 64 bit hash and compact its entropy in 32 bits.
+     * @deprecated This function should only be used for keeping backward compatibility with external libraries using 32 hashes.
```

Comment:
If I understand that correctly, nothing external relies on shape of presto hashes. However, Fastutil Hashset requires hash providers (HashStrategies) and our implementation uses hashing function to provide them. This is why this function is required to provide them without changing unrelated code right now.

Even more tricky point is using 64 bit hashes in InCodeGenerator (switch varian) as java only support switches on integers, without such fallback we will have to redesign whole generator to avoid switches.